impeller/renderer/backend/metal/allocator_mtl.mm

```diff
@@ -88,6 +90,18 @@ static bool SupportsLossyTextureCompression(id<MTLDevice> device) {
 #endif
 }
 
+void DebugAllocatorStats::Increment(size_t size) {
+  size_.fetch_add(size, std::memory_order_relaxed);
+}
+
+void DebugAllocatorStats::Decrement(size_t size) {
+  size_.fetch_sub(size, std::memory_order_relaxed);
```

Comment:
Do you think we should have an assert that size will not go below zero?

Reply:
Well since its an unsigned type, it won't go below zero 🤓

Reply:
Semantics! haha.  It'll go below zero, but you won't like it.